### ISSUE TITLE
Help regarding usage

### ISSUE THREAD

**Issue by CybernetHacker14** [zlib-ng/minizip-ng#688]
I am developing a basic API for reading from and writing to a zip file. Currently, I am testing this both with `zlib` and `minizip-ng` libraries.
Here's the code for it:

```cpp
bool ZipFileOperations::ReadFromZipFile(
  const std::string & zipPath,
    const std::string & inZipLocation, void ** ptrToBuffer) {
  unzFile zf = unzOpen(zipPath.c_str());

  if (zf == NULL) {
    ST_ASSERT(false, "Error opening zip file");
    return false;
  }

  if (unzGoToFirstFile(zf) == UNZ_OK) {
    do {
      if (unzOpenCurrentFile(zf) == UNZ_OK) {
        unz_file_info fileInfo;
        memset( & fileInfo, 0, sizeof(unz_file_info));

        if (unzGetCurrentFileInfo(zf, & fileInfo, NULL, 0, NULL, 0, NULL, 0) == UNZ_OK) {
          char * filename = (char * ) malloc(fileInfo.size_filename + 1);
          unzGetCurrentFileInfo(zf, & fileInfo, filename, fileInfo.size_filename + 1, NULL, 0, NULL, 0);
          filename[fileInfo.size_filename] = '\0';
          std::string path(filename);

          if (path == inZipLocation) {
            * ptrToBuffer = new char[fileInfo.uncompressed_size];
            unzReadCurrentFile(zf, * ptrToBuffer, fileInfo.uncompressed_size);
            free(filename);
            return true;
          }

          free(filename);
        }

        unzCloseCurrentFile(zf);
      }
    } while (unzGoToNextFile(zf) == UNZ_OK);
  }

  return false;
}

bool ZipFileOperations::WriteToZipFile(
  const std::string & zipPath,
    const std::string & inZipLocation,
      const char * data, UInt32 length) {
  int appendMode;

  zipFile zf = zipOpen(
    zipPath.c_str(),
    !Filesystem::DoesFileExist(zipPath) || DoesFileExistInZip(zipPath, inZipLocation) ? APPEND_STATUS_CREATE :
    APPEND_STATUS_ADDINZIP);

  if (zf == NULL) return false;

  zip_fileinfo zfi = {
    0
  };

  if (Z_OK != zipOpenNewFileInZip(
      zf,
      inZipLocation.c_str(), &
      zfi,
      NULL,
      0,
      NULL,
      0,
      "",
      Z_DEFLATED, // - This parameter here, fucking POS
      Z_BEST_COMPRESSION)) { // Compression level (9=max)
    return false;
  }

  if (Z_OK != zipWriteInFileInZip(zf, length == 0 ? "" : data, length)) return false;

  if (Z_OK != zipCloseFileInZip(zf)) return false;

  if (zipClose(zf, NULL)) return true;

  return false;
}
```

I am performing the following steps - 
1. Created new zip. Added a new file to it named `Test.scene`.
2. Edited the contents of `Test.scene` file - basically edited a string present in the file.
3. Created a new file in the zip named `Test2.scene`.
4. Edited the contents of `Test2.scene` file - basically edited a string present in the file

Basically, when I performed the fourth step, the first file disappears. I am not able to figure out why is this happening.
I tried combing through the [docs](https://github.com/zlib-ng/minizip-ng/tree/master/doc), to find out exactly how to use the library in a proper way, since I am new to this field of zip archive manipulation - and trying to just pack stream data into a zip format.

Can someone help me with this please?

**Comment by nmoinvaz** [comment]
Zip format does not handle in-place update of zip entries well. You must first do something like `minizip_erase` to erase the existing entry and then you can add new entry.

**Comment by nmoinvaz** [comment]
Since your code uses the minizip compatible API it might better to ask on StackOverflow for support if you are still having problems. If you find out it is indeed a bug then please reopen.
